File: BinaryTreePaths/BinaryTreePaths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class BinaryTreePaths {
    /* dude this is beautiful recursion owo
     *
     */
    public List<String> binaryTreePaths(TreeNode root) {
        if (root == null) {
            return new ArrayList<String>();
        } else {
            List<String> paths = binaryTreePaths(root.right);
            paths.addAll(binaryTreePaths(root.left));
            if (paths.isEmpty()) {
                paths.add(Integer.toString(root.val));
                
            } else {
               paths = paths.stream().map(x -> Integer.toString(root.val) + "->" + x).collect(Collectors.toList());
            }
            return paths;
        }
        
    }
}

File: BinaryTreePaths/BinaryTreepaths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def binaryTreePaths(self, root: TreeNode) -> List[str]:
        paths = []
        if root == None: return paths
        def traverse(self, root, prev):
            nonlocal paths
            if (root == None):
                return
            elif (root.right == None and root.left == None):
                paths += [prev + str(root.val)]
            else:
                traverse(self, root.right, prev + str(root.val) + "->")
                traverse(self, root.left, prev + str(root.val) + "->")
        
        traverse(self, root, "")
        return paths

        
